### PR TITLE
Base64 keywords documentation v1

### PR DESCRIPTION
**Pull request by Syoc**
Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [X ] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [X ] I have signed the Open Information Security Foundation contribution agreement at https://suricata-ids.org/about/contribution-agreement/
- [X ] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/3035

Describe changes:
- I have added documentation for base64-decode and base64-content
-
-

[PRScript](https://redmine.openinfosecfoundation.org/projects/suricata/wiki/PRscript) output (if applicable):

### CODE REVIEW

File: doc/userguide/rules/base64-keywords.rst [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+Syntax::
+
+    base64_decode:bytes <value>, offset <value>, relative;
+
+The ``bytes`` option specifies how many bytes that should be decoded and made available for base64_data.
+The decoding will stop at the end of the buffer.
+
+The ``offset`` option specifies how many bytes to skip before the decoding should start. Bytes are skipped relative to the start of the payload buffer if relative is not set.
+
+The ``relative`` option makes the decoding start relative to the previous content match. Default behavior is to start at the beginning  of the buffer. This option makes ``offset`` skip bytes relative to the previous match.
```
Comment (norg) [inline doc/userguide/rules/base64-keywords.rst]: between beginning and of the buffer are double spaces.
It might also be better to have a line break.

**Comment by victorjulien** [comment]
Could you do a new PR with all commits squashed into a single commit. See https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Code_Submission_Quality_Criteria#Commits

File: doc/userguide/rules/base64-keywords.rst (hunk 1 above)
Comment (Syoc) [reply doc/userguide/rules/base64-keywords.rst]: I'm sorry, but I don't fully understand the first sentence.
I'll line breaks for my next pull request.

**Comment by Syoc** [comment]
Thank you both for having a look. I'll open a revised pull request later.

File: doc/userguide/rules/base64-keywords.rst (hunk 1 above)
Comment (norg) [reply doc/userguide/rules/base64-keywords.rst]: between "Default behavior is to start at the beginning" and "of the buffer." are two spaces and not just one.

**Pull request closed without merging**